app/site/controllers.py

```diff
@@ -31,23 +31,14 @@ def index():
       200:
         description: Succesfuly loaded home page
     """
-    try:
-        if request.method == "POST":
-            encoded_token = request.form.get('encoded_token', '')
-            if encoded_token:
-                try:
-                    payload = jwt.decode(encoded_token, app.config['API_KEY'])
-                except Exception as e:
-                    app.logger.error(e)
-                    return redirect(request.headers['Host'] + '/logout')
-                user = User().get_userinfo_by_id(payload['user'])
-                if user:
-                    return render_template("dashboard.html", user=user,
-                                           title='Dashboard'), 200
-                return redirect(request.headers['Host'] + '/logout')
-        return render_template("index.html", title='Home'), 200
-    except Exception:
-        return redirect(url_for('.logout'))
+    user, exception = get_user_from_cookie()
+    if user:
+        return render_template("dashboard.html",
+                               user=user,
+                               title='Dashboard'), 200
+    if exception:
+        return redirect(request.headers['Host'] + '/logout')
```

Comment:
@rufuspollock redirect(url_for(logout)) takes `API GATEWAY` url. As we are redirecting from dreamhost to apigateway.
So redirect(url_for(logout)) would take `aws.dpr-api-{hash}/stage`